eras/allegra/impl/testlib/Test/Cardano/Ledger/Allegra/CDDL.hs

```diff
@@ -46,44 +50,41 @@ native_script =
         |    specifies the right (excluded) endpoint b.
         |]
     $ "native_script"
-      =:= arr [a script_pubkey]
-      / arr [a script_all]
-      / arr [a script_any]
+      =:= arr [a allegra_script_pubkey]
+      / arr [a allegra_script_all]
+      / arr [a allegra_script_any]
       / arr [a script_n_of_k]
-      / arr [a invalid_before]
-      / arr [a invalid_hereafter]
+      / arr [a allegra_invalid_before]
+      / arr [a allegra_invalid_hereafter]
 
-script_pubkey :: Named Group
-script_pubkey = "script_pubkey" =:~ grp [0, a addr_keyhash]
+allegra_script_pubkey :: Named Group
+allegra_script_pubkey = "script_pubkey" =:~ grp [0, a addr_keyhash]
 
-script_all :: Named Group
-script_all = "script_all" =:~ grp [1, a (arr [0 <+ a native_script])]
+allegra_script_all :: Named Group
+allegra_script_all = "script_all" =:~ grp [1, a (arr [0 <+ a allegra_native_script])]
 
-script_any :: Named Group
-script_any = "script_any" =:~ grp [2, a (arr [0 <+ a native_script])]
+allegra_script_any :: Named Group
+allegra_script_any = "script_any" =:~ grp [2, a (arr [0 <+ a allegra_native_script])]
 
 script_n_of_k :: Named Group
-script_n_of_k = "script_n_of_k" =:~ grp [3, "n" ==> int64, a (arr [0 <+ a native_script])]
+script_n_of_k = "script_n_of_k" =:~ grp [3, "n" ==> int64, a (arr [0 <+ a allegra_native_script])]
```

Comment:
I wonder if we could parameterize these rules by the native_script rule, and then use them in each era  with the respenctive native_script definition